server/route.go

```diff
@@ -491,6 +481,20 @@ func routeSid(sub *subscription) string {
 	return fmt.Sprintf("%s%s:%d:%s", qi, RSID, sub.client.cid, sub.sid)
 }
 
+func parseRouteSid(rsid []byte) (cid uint64, sid []byte, ok bool) {
+	if !bytes.HasPrefix(rsid, []byte(QRSID_PREFIX)) {
+		return
+	}
+
+	for i := QRSID_PREFIX_LEN; i < len(rsid); i++ {
+		switch rsid[i] {
+		case ':':
+			return uint64(parseInt64(rsid[QRSID_PREFIX_LEN:i])), rsid[i+1:], true
```

Comment:
I apologize, you are right, since `i` will always be `<len(rsid)`, reslicing at `i+1` would simply return an empty slice in that case.

Although your code is now equivalent in behavior with the regex, I now realize that we may have a problem and need to discuss this with @derekcollison. It looks to me that as long as we have a `QRSID` prefix, we should *always* return ok, no matter if we are able to decode cid and/or sid.

My reasoning is that if we return !ok when for instance getting `QRSID:1 2\r\nok\r\n` like in one of your test, the server ends up sending to a regular subscriber. I am not sure this is the original intent. As you can see in [here](https://github.com/nats-io/gnatsd/blob/6a12c218cd5e6c82a207fb979b1775e5df52aedd/server/client.go#L1117), it looks like if the message was for a queue subscription, the server should not attempt to deliver to other subs, even if no actual queue subscriber is found (line 1124).

Let me check with Derek and once we agree we will get back to you (if parseRouteSid should return ok as long as the prefix is found).